test/common/test_context.go

```diff
@@ -754,6 +754,19 @@ func (ctx *TestContext) RegisterPlatform() *types.Platform {
 	return ctx.RegisterPlatformWithType("test-type")
 }
 
+func (ctx *TestContext) RegisterTenantPlatform() *types.Platform {
```

Comment:
did you see `common.GenerateRandomPlatform()`? not sure we need this as well

Reply:
we can use RegisterPlatformInSM(GenerateRandomPlatform(), ctx.SMWithOAuthForTenant, map[string]string{}) directly, but this seems to be more readble this way?

Reply:
you can still use the other one as the impl. of this one, no need for the logic again.